Cudasim tests will fail when NumPy 1.20 is added to the test matrix

Not quite an issue yet as I don't think we officially support NumPy 1.20, but recording this here as it will need resolving. Running the CUDA testsuite with cudasim and NumPy 1.20 gives a couple of failures:

```
======================================================================
FAIL: test_incompatible_highdim (numba.cuda.tests.cudadrv.test_cuda_array_slicing.CudaArraySetting)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gmarkall/numbadev/numba/numba/cuda/tests/cudadrv/test_cuda_array_slicing.py", line 289, in test_incompatible_highdim
    self.assertIn(
AssertionError: 'could not broadcast input array from shape (2,3) into shape (35,)' not found in ["Can't assign 3-D array to 1-D self", 'could not broadcast input array from shape (2,3) into shape (35)']

======================================================================
FAIL: test_incompatible_shape (numba.cuda.tests.cudadrv.test_cuda_array_slicing.CudaArraySetting)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/gmarkall/numbadev/numba/numba/cuda/tests/cudadrv/test_cuda_array_slicing.py", line 303, in test_incompatible_shape
    self.assertIn(
AssertionError: 'could not broadcast input array from shape (2,) into shape (5,)' not found in ["Can't copy sequence with size 2 to array axis 0 with dimension 5", 'cannot copy sequence with size 2 to array axis with dimension 5']

----------------------------------------------------------------------
```

IIRC NumPy 1.20 will be supported for 0.54, so I'll put this in the 0.54 milestone.

--- a/numba/cuda/tests/cudadrv/test_cuda_array_slicing.py
+++ b/numba/cuda/tests/cudadrv/test_cuda_array_slicing.py
@@ -291,7 +291,9 @@ class CudaArraySetting(CUDATestCase):
             container=[
                 "Can't assign 3-D array to 1-D self",  # device
                 "could not broadcast input array from shape (2,3) "
-                "into shape (35)",  # simulator
+                "into shape (35)",  # simulator, NP < 1.20
+                "could not broadcast input array from shape (2,3) "
+                "into shape (35,)",  # simulator, NP >= 1.20
             ])
 
     def test_incompatible_shape(self):
@@ -306,7 +308,9 @@ class CudaArraySetting(CUDATestCase):
                 "Can't copy sequence with size 2 to array axis 0 with "
                 "dimension 5",  # device
                 "cannot copy sequence with size 2 to array axis with "
-                "dimension 5",  # simulator
+                "dimension 5",  # simulator, NP < 1.20
+                "could not broadcast input array from shape (2,) into "
+                "shape (5,)",   # simulator, NP >= 1.20
             ])
 
     @skip_on_cudasim('cudasim does not use streams and operates synchronously')

